docs/upgrade-notes.asciidoc

```diff
@@ -0,0 +1,1714 @@
+[[breaking-changes-summary]]
+== Upgrade notes
+
+////
+USE THE FOLLOWING TEMPLATE to add entries to this document.
+1. Copy from "[discrete]" to the last "====" included.
+2. Paste and edit the template in the right area section. Most recent entries should be at the top of the section. 
+3. Make sure to edit the anchor ID [[REPO-PR]] with proper values, then add the corresponding entry to the doc link service https://github.com/elastic/kibana/blob/main/packages/kbn-doc-links/src/get_doc_links.ts:
+
+codeId: `${KIBANA_DOCS}breaking-changes-summary.html#REPO-PR` // replace `codeId` with the ID of your choice that you want to call in the Upgrade Assistant code, and replace `REPO-PR` with the corresponding values you edited in this document.
+
+[discrete]
+[[REPO-PR]]
+.[FEATURE] TITLE TO DESCRIBE THE CHANGE. (VERSION)
+[%collapsible]
+====
+*Details* +
+ADD MORE DETAILS ON WHAT IS CHANGING AND A LINK TO THE PR INTRODUCING THE CHANGE
+
+*Impact* +
+ADD INFORMATION ABOUT WHAT THIS CHANGE WILL BREAK FOR USERS
+
+*Action* +
+ADD INSTRUCTIONS FOR USERS LOOKING TO UPGRADE. HOW CAN THEY WORK AROUND THIS?
+====
+////
```

Comment:
@lcawl @jloleysens @rudolf Here is the file where we can compile breaking changes and deprecation announcements at the same place. Based on our discussion yesterday, can you let me know your thoughts on these 2 things:
- The content template just here at the top of the file. We could ask teams to update this file following that template when adding deprecations or breaking changes. In that template are instructions on what to do to populate links in the Upgrade Assistant through the doc links service file.
- The IDs for each item. Right now they are not entirely consistent with each other because they were compiled from existing sources, but for each new addition to this file, we can use a fixed pattern, such as `[[kibana-PR_number]]` as proposed in the template.

Reply:
That all seems like a good step forward as we ultimately get move toward more consistency and automation across our release docs.

Reply:
This looks like a great start!